README.md

```diff
@@ -2,25 +2,60 @@
 
 Smart contracts for [bnomial](https://github.com/underfitted/bnomial).
 
+## Repository Organization
+
+The main artifacts (files and folders) in the repository are:
+
+- [contracts](contracts/): It contains smart contracts, NFTs and related utility functions for the project;
+- [scripts](scripts/): [Hardhat](https://hardhat.org/) custom scripts;
+- [test](test/): Bnomial test sets;
+- [hardhat.config.js](hardhat.config.js): Main configuration file for Hardhat. You should customize it if you want to deploy
+the contracts in remote networks.
+
 ## Setup
 
-To setup your environment:
+### Local machine \ IDE agnostic
+
+You should have [npm](https://www.npmjs.com/) installed, so be sure to follow the proper procedure for your OS. After that, download and install all required libraries and plugins by running:
 
 ```
 npm install
 ```
 
-To setup your API keys and private keys create an `.env` file with the following content (the `POLYGON_API_URL` is not needed if you are just testing on Mumbai).
+### VSCode Remote Container
 
-```
-MUMBAI_API_URL="<Alachemy HTTP URL>"
-POLYGON_API_URL="<Alachemy HTTP URL>"
-PRIVATE_KEY="<deploy wallet private key>"
-POLYSCAN_KEY="<Ploygonscan API key for verification>"
-```
+You may use [VSCode Remote Container](https://code.visualstudio.com/docs/remote/containers) if you want to avoid local
+installations. Just install the extension and setup your local Docker as described 
+[here](https://code.visualstudio.com/docs/remote/containers#_getting-started).
+
+When running the project in the development container, you may use the embedded vscode terminal to execute the 
+_deployment workflow_.
+
+> :warning: The container runs all processes and update the files as the internal user _node:node_ (1000:1000).
 
 ## Develop
 
+Smart contracts and NFTs demand an account and a blockchain network to be deployed into. Besides that, it's important to set an
```

Comment:
Thanks for the detailed docs. We already have some detailed description of how to set up the environment here, but I think we forgot to link it in the README... 

https://docs.underfitted.io/incubator/bnomial/web3/environment-setup

Reply:
Yeah... saw that. The doc there is much more detailed. I put this here because I felt like something was missing and a little bit of explanation would help when building the stuff. So, should I remove and just mention the link to the docs?

Reply:
We could have a line in the beginning of Set Up section like this:

> There is a detailed description of Bnomial project and some guidelines regarding its set up and development. You can find them in this [link](https://docs.underfitted.io/incubator/bnomial/web3/environment-setup).

What do you think?

Reply:
Yes, I like that. Let's keep the rest of the info you wrote here. We just need to be aware that both sources may get out of sync at some point.

Reply:
Added this text further down.